extractor/src/main/scala/org/jetbrains/sbt/extractors/ProjectExtractor.scala

```diff
@@ -306,10 +315,16 @@ object ProjectExtractor extends SbtStateOps with TaskOps {
         taskInCompile(Keys.scalaInstance).onlyIf(options.download).value
       val scalaCompilerBridgeBinaryJar =
         keysAdapterEx.myScalaCompilerBridgeBinaryJar.value
-      val scalacOptions =
-        taskInCompile(Keys.scalacOptions).onlyIf(options.download).value
-      val javacOptions =
-        taskInCompile(Keys.javacOptions).onlyIf(options.download).value
+
+      val configurationToScalacOptions = Map(
```

Comment:
It would be very interesting to investigate the performance impact of this.
Please write down this task as a separate step to do.
If `key.in(projectRef, config).get(state)` operation has a big "warmup" overhead, the performance might downgrade.

Reply:
I didn't expect a long warmup here, but ofc I will check.

Reply:
I made tests with `lila`, `zio` and `cats`. I haven't noticed any downgrade. 
`cats`
**old** 2s;2s  
**new**  3s;2s
`zio`
**old** 4s 
**new**  3s;5s
`lila`
**old** 5s 
**new**  5s

Reply:
Ok, good